src/mpid/ch4/subconfigure.m4

```diff
@@ -467,6 +467,16 @@ if test "$enable_ch4r_per_comm_msg_queue" = "yes" ; then
         [Define if CH4U will use per-communicator message queues])
 fi
 
+AC_ARG_WITH(ch4-max-vcis,
+    [--with-ch4-max-vcis=<N>
```

Comment:
How does this tie with max vni, e.g., MPIR_CVAR_CH4_OFI_MAX_VNIS?

Reply:
They are not tied explicitly. `MPIDI_CH4_MAX_VCIS` defines maximum logical VCIs. `MPIR_CVAR_CH4_OFI_MAX_VNIS` defines maximum "physical" netmod VNIs. When they are not equal, the netmod may define a mapping scheme. Of course, to keep it simple, netmod may simply choose the smaller of them.

Reply:
If a VNI is a physical resource that is not exposed outside of the netmod, I question the name of "virtual network interface". Can we use EP or SEP directly?

Reply:
Yepp, you are right.

Reply:
> If a VNI is a physical resource that is not exposed outside of the netmod, I question the name of "virtual network interface". Can we use EP or SEP directly?

We could. Ultimately, it is each device module's decision and based on what fits. In the example OFI, we can either use EP inside `fi_domain` or context inside `SEP`, so a abstract name such VNI may still make sense.

Reply:
Just a note: for performance, we probably would like the `VCI` to `VNI` mapping be one-to-one. We could use an assert (inside ofi_init) to ensure this condition.